### PR TITLE
Ensure Terraform installer can be re-run

### LINKED ISSUE #176: Ensure Terraform installer can be re-run
For future updates, we need to ensure that the Terraform updater can be re-run without breaking existing deployed resources (it should work... it's terraform; but let's confirm).

Also provide docs about terraform install and update.

### PR DESCRIPTION
**Pull request by davidstanke**
Fixes #176 

This PR has a handful of changes, mostly preparing the README for the terraform installer to be ready to grow up and be the real setup README.

### CODE REVIEW

**Comment by davidstanke** [comment]
(Oh and btw, yes, re-running terraform works as expected)

File: experimental/terraform/README.md [hunk 1]
```diff
@@ -1,42 +1,187 @@
...
+> installation method (hopefully soon!): 
+> * move the current contents of `/setup` to `/setup/deprecated`
+> * move the contents of this directory to `/setup`
+> * remove this note
+
+# Installation guide
+
+This guide describes how to set up Four Keys with your GitHub or GitLab project. The main steps are:
+
+1.  [Runing the setup script](#running-the-setup-script)
```
Comment (dinagraves) [inline experimental/terraform/README.md:12]:
```suggestion
1.  [Running the setup script](#running-the-setup-script)
```

File: experimental/terraform/README.md [hunk 2]
```diff
@@ -1,42 +1,187 @@
...
+1.  Integrating with your GitHub or Git Lab repo by:
```
Comment (dinagraves) [inline experimental/terraform/README.md:13]:
```suggestion
1.  Integrating with your GitHub or GitLab repo by:
```

File: experimental/terraform/README.md [hunk 3]
```diff
@@ -1,42 +1,187 @@
...
+    1.  [Collecting changes data](#collecting-changes-data)
+    1.  [Collecting deployment data](#collecting-deployment-data)
+    1.  [Collecting incident data](#collecting-incident-data)
+
+## Before you begin
+
+1.  Install [GCloud SDK](https://cloud.google.com/sdk/install).
+1.  You must be owner on a Google Cloud project that has billing enabled. You may either use this project to house the architecture for the Four Keys, or you will be given the option to create new projects. If you create new projects, the original Google Cloud project will NOT be altered during set up, but the billing information from this parent project will be applied to any projects created.
```
Comment (dinagraves) [inline experimental/terraform/README.md:21]: Should we make the billing information bold?  Important piece of information.

File: experimental/terraform/README.md [hunk 4]
```diff
@@ -1,42 +1,187 @@
...
+    *   Would you like to create a new Google Cloud Project for the four key metrics? (y/N)
+        * If you choose no, you will be asked later to provide info about the project you wish to use.
+    *   Choose the event sources to configure...
+        *   Which version control system are you using?
+            * Choose the appropriate option for your VCS, or choose "other" to skip VCS integration
+        * Which CI/CD system are you using?
+            * Choose the appropriate option for your CICD system, or choose "other" to skip CICD integration
+        * _(see `/README.md#extending-to-other-event-sources` to integrate event sources not available during setup)_
+    *   Would you like to generate mock data? (y/N)
+        *   If you select yes, a script will run through and send mock GitLab or GitHub events to your event-handler.  This will populate your dashboard with mock data.  The mock data will include the work "mock" in the source. You can generate mock data without using the setup script. See [Generating mock data](../readme.md).
```
Comment (dinagraves) [inline experimental/terraform/README.md:45]: To exclude the mockdata from the dashboard, update the SQL script to filter out any source with the word mock in it.   `WHERE source not like "%mock"`

File: experimental/terraform/README.md [hunk 5]
```diff
@@ -1,42 +1,187 @@
...
+
+Step by step, here's what's happening:
+1. `setup.sh` starts by collecting information from the system and the user to determine a number of configuration variables that will be provided to Terraform.
+1. If you choose to make a new project, `setup.sh` will use `gcloud` to create that project, with an ID in the form `fourkeys-XXXX`.
+1. It sets several environment variables, and writes a `terraform.tfvars` file to disk, containing inputs to Terraform.
+1. Then it invokes `install.sh`, which is responsible for provisioning the infrastructure.
+1. `install.sh` runs `gcloud builds submit` commands to build the application containers that will be used in Cloud Run services.
+1. Then it invokes Terraform, which processes the configuration files (ending in `.tf`) to provision all of the necessary infrastructure into the speficied Cloud project.
+1. If you've chosen to generate mock data, the script then calls the "data generator" python application to submit several synthetic webhook events to the event-handler service that was just created.
```
Comment (dinagraves) [inline experimental/terraform/README.md:68]: Maybe link to the generate_data.py file?

**Review by dinagraves — APPROVED** [review]

File: experimental/terraform/README.md (hunk 3 above)
Comment (davidstanke) [reply experimental/terraform/README.md]: done

File: experimental/terraform/README.md (hunk 4 above)
Comment (davidstanke) [reply experimental/terraform/README.md]: added!

File: experimental/terraform/README.md (hunk 5 above)
Comment (davidstanke) [reply experimental/terraform/README.md]: linked!

**Pull request merged**